matrix/plugins/glitch/actions.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+import logging
+from typing import Any
+from functools import wraps
+
+import enforce
+from juju.model import Model
+from juju.unit import Unit
+
+from matrix.utils import Singleton
+
+log = logging.getLogger("glitch")
+
+
+class _Actions(dict, metaclass=Singleton):
+    """
+    Interface for registering actions, translating text strings into
+    action functions, and performing automagic around our action
+    functions.
+
+    """
+    def decorate(self, func):
+        """
+        Register an action. Return a function that accepts a set of objects,
+        and iterates over that set, running the registered action on each
+        object in that set.
+
+        """
+        @enforce.runtime_validation
+        @wraps(func)
+        async def wrapped(model, objects, **kwargs):
+            for obj in objects:
+                await func(model, obj, **kwargs)
```

Comment:
That's a fair point.  I'm fine with sticking with the current approach to ease single-point-of-handling of those options.  That functionality is clearly not compatible with the actions handling multiple items themselves.